.github/workflows/test.yml

```
@@ -0,0 +1,72 @@
+name: Haskell CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  cabal:
+    name: ghc ${{ matrix.ghc }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+        ghc:
+          - 8.10.4
+        cabal:
+          - latest
+
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_PASSWORD: postgres
+        # Set health checks to wait until postgres has started
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 5432:5432
+
+    steps:
+    - name: Install dependencies
+      run: |
+        sudo apt-get install libkrb5-dev libpq-dev libssl-dev
+
+    - uses: actions/checkout@v2
+
+    - uses: haskell/actions/setup@v1
+      id: setup-haskell-cabal
+      name: Setup cabal
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: ${{ matrix.cabal }}
+
+    - name: Configure cabal
+      run: |
+        cabal configure --enable-tests --enable-benchmarks
+    - uses: actions/cache@v2
+      name: Cache cabal artifacts
+      with:
+        path: |
+          ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+          dist-newstyle
+        key: ${{ runner.os }}-cabal-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+        restore-keys: ${{ runner.os }}-cabal-${{ matrix.ghc }}-
+
+    - name: Install dependencies
+      run: |
+        cabal update
+        cabal build --only-dependencies --enable-tests --enable-benchmarks
+    - name: Build
+      run: cabal build --enable-tests --enable-benchmarks all
+
+    - name: Run tests
+      env:
+        DATABASE_URL: postgres://postgres:postgres@localhost:5432/postgres
+      run: cabal test all
```

```suggestion
      run: cabal test --enable-tests --enable-benchmarks all
```
I tried this locally and it doesn't seem to trigger rebuilds.

I've got a suspicion that `cabal build --enable-tests --enable-benchmarks` constrains the `cabal-install` solver differently from _just_ `cabal test`.

This shouldn't matter when using a freeze file, though, which has me a bit confused.